amazon-sponsored-items-blocker.user.js

```diff
@@ -28,12 +30,10 @@ const isAd = (node) =>
 
 /**
  * @param {Node[]} nodes
+ * @param {function(Node|Element): boolean} pred
  */
-const removeSponsoredAds = (nodes) => {
-  /**
-   * @type {HTMLElement[]}
-   */
-  const ads = nodes.filter(isAd);
+const removeSponsoredAds = (nodes, pred) => {
+  const ads = nodes.filter(pred);
```

Comment:
参照したりカウントしたりするから、やっぱ必要は必要。

ただこうしてもいい

```suggestion
  const ads = Array.from(nodes).filter(pred);
```